include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -7491,6 +7634,13 @@ FEFaceEvaluation<dim,
   if (!(evaluate_values + evaluate_gradients))
     return;
 
+  const unsigned int face_no =
+    (this->dof_access_index ==
+       internal::MatrixFreeFunctions::DoFInfo::dof_access_cell &&
+     this->is_interior_face == false) ?
+      compute_face_no_data()[0] :
+      this->face_no;
+
```

Comment:
@kronbichler I have modified the code so that the function `compute_face_no_data` is only called for exterior faces in the case of ECL. At the moment, I am only using the information of the first lane, but in a next PR I would also use the information of other lanes to be able to support unstructured meshes.